### PR TITLE
FIX: composite part name editing (ISX-1734)

### PR DESCRIPTION
**Pull request by ritamerkl**
### Description

This PR addresses the issue that composite parts where not editable and operate the dropdown menu changed the binding path instead. The ticket can be found [here](https://jira.unity3d.com/browse/ISX-1734). 

### Changes made

Fixed:
- editing composite parts
The problem was that the invoked command changed the m_Path of the binding and not like it shout the m_Name property.

### Checklist

Before review:

- [x] Changelog entry added.
    - Explains the change in `Changed`, `Fixed`, `Added` sections.
    - For API change contains an example snippet and/or migration example.
    - FogBugz ticket attached, example `([case %number%](https://issuetracker.unity3d.com/issues/...))`.
    - FogBugz is marked as "Resolved" with *next* release version correctly set.
- [ ] Tests added/changed, if applicable.
    - Functional tests `Area_CanDoX`, `Area_CanDoX_EvenIfYIsTheCase`, `Area_WhenIDoX_AndYHappens_ThisIsTheResult`.
    - Performance tests.
    - Integration tests.
- [ ] Docs for new/changed API's.
    - Xmldoc cross references are set correctly.
    - Added explanation how the API works.
    - Usage code examples added.
    - The manual is updated, if needed.

During merge:

- [ ] Commit message for squash-merge is prefixed with one of the list:
    - `NEW: ___`.
    - `FIX: ___`.
    - `DOCS: ___`.
    - `CHANGE: ___`.
    - `RELEASE: 1.1.0-preview.3`.

### CODE REVIEW

**Review by Pauliusd01 — APPROVED** [review]
LGTM. Checked that the Composite Part option no longer messes with the path on different composite types. (Some, I was unable to check due to implementation issues like ISX-1817, but ones that are available work correctly now)

File: Packages/com.unity.inputsystem/CHANGELOG.md [hunk 1]
```diff
@@ -19,6 +19,7 @@ however, it has to be formatted properly to pass verification tests.
 - Fixed `Destroy may not be called from edit mode` error [ISXB-695](https://issuetracker.unity3d.com/product/unity/issues/guid/ISXB-695)
 - Fixed possible exceptions thrown when deleting and adding Action Maps.
 - Fixed potential race condition on access to GCHandle in DefferedResolutionOfBindings and halved number of calls to GCHandle resolution [ISXB-726](https://issuetracker.unity3d.com/product/unity/issues/guid/ISXB-726)
+- Fixed composite parts not editable issue
```
Comment (ekcoh) [inline Packages/com.unity.inputsystem/CHANGELOG.md]: Could you rewrite this changelog item to provide more clarity on what previous symptom it resolves, it is quite unclear what it addresses in its current form. Also missing full-stop in the end.

File: Packages/com.unity.inputsystem/InputSystem/Editor/Internal/InputActionSerializationHelpers.cs [hunk 2]
```diff
@@ -395,6 +395,12 @@ public static void DeleteActionAndBindings(SerializedProperty actionMap, Guid ac
             return newBindingProperty;
         }
 
+        public static void SetBindingPartName(SerializedInputBinding bindingProperty, string partName)
+        {
+            //expects nicified part name
```
Comment (ekcoh) [inline Packages/com.unity.inputsystem/InputSystem/Editor/Internal/InputActionSerializationHelpers.cs]: Typo in comment

**Review by ekcoh — CHANGES_REQUESTED** [review]

**Review by ekcoh — APPROVED** [review]
LGTM - approved, needs update from develop and CI rerun it seems.

**Pull request merged**